GH-6375: Harmonize package structures

artembilan opened the pull request:
Fixes: https://github.com/spring-projects/spring-integration/issues/6375

* Move respective classes to the `inbound`, `outbound` or `support` packages to make the whole picture for the API more concise and clean

The firs commit is for `stream` module

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

artembilan committed:
GH-6375: Harmonize package structures

Fixes: https://github.com/spring-projects/spring-integration/issues/6375

* Move respective classes to the `inbound`, `outbound` or `support` packages
to make the whole picture for the API more concise and clean

The firs commit is for `stream` module

artembilan committed:
Fix `stream.adoc` for new `inbound/outbound` packages

artembilan committed:
Hormonize packages for FTP & SFTP modules

artembilan committed:
Hormonize packages for WS module

* Improve `ws.adoc` to avoid `foo`, `bar` etc.

artembilan committed:
Harmonize packages in `mail` module

* Move inbound channel adapters into a new `inbound` package
* Move outbound channel adapters into a new `outbound` package
* Preserve existing classes in the root package as deprecated
and as extensions of new classes

artembilan committed:
Harmonize packages in `jmx` module

* Move inbound channel adapters into a new `inbound` package
* Move outbound channel adapters into a new `outbound` package
* Preserve existing classes in the root package as deprecated
and as extensions of new classes
* Some tests refactoring according to modern AssertJ API
* Fix typos in `jmx.adoc`

artembilan committed:
Harmonize packages in `file` module

* Move inbound channel adapters into a new `inbound` package
* Move outbound channel adapters into a new `outbound` package
* Preserve existing classes in the root package as deprecated
and as extensions of new classes
* Some tests refactoring (affected files) according to modern AssertJ API

artembilan committed:
Harmonize packages in `jms` module

* Move inbound channel adapters into a new `inbound` package
* Move outbound channel adapters into a new `outbound` package
* Move channels into a new `channel` package
* Preserve existing classes in the root package as deprecated
and as extensions of new classes
* Some tests refactoring (affected files) according to modern AssertJ API
* Use now a new in SF `SimpleDestinationResolver` for optimization with caching

cppwfs reviewed (changes requested):
What a PR!   😆 

Great work!   Breaking up the inbound and outbound classes into their own packages makes perfect sense. 

Just my usual nitpicks.

artembilan commented on spring-integration-stream/src/main/java/org/springframework/integration/stream/event/StreamClosedEvent.java:
```diff
@@ -0,0 +1,36 @@
+ * Application event published when EOF is detected on a stream.
+ *
+ * @author Gary Russell
+ *
```
This was just a class move.
I add (and request) `@author` tag when a reasonable change is made to the class.

artembilan commented on spring-integration-file/src/main/java/org/springframework/integration/file/FileReadingMessageSource.java:
```diff
@@ -94,48 +63,16 @@
 	 * Create a FileReadingMessageSource with a naturally ordered queue of unbounded capacity.
 	 */
 	public FileReadingMessageSource() {
-		this(null);
```
Why?
If we don't specify `super()`, that `super()` is compiled-in.
And that `super()` does exactly what you are asking.

artembilan commented:
> do the moving in one commit and re-introduce the now deprecated classes as new ones

That is great idea!
I don't think I'll revert anything what I have done so far, but I'll try to follow your recommendations for two remaining modules.

Thank you for review!

> in all previous tags and branches with their complete history

Yeah... Unfortunately, after removing deprecated classes in the next version we won't have their mentioning anywhere to follow the history.

artembilan committed:
Fix XSDs for proper class packages

cppwfs commented on spring-integration-ws/src/main/java/org/springframework/integration/ws/MarshallingWebServiceOutboundGateway.java:
```diff
@@ -103,59 +99,10 @@ public MarshallingWebServiceOutboundGateway(@Nullable String uri, WebServiceTemp
-		public void doWithMessageInternal(WebServiceMessage message, Object payload) throws IOException {
-		}
 
+		super(destinationProvider, webServiceTemplate);
```
In the code review, I really didn't see an appreciable benefit to adding them.  
I think the rule is, if you add anything outside `super(...)` then a test would be required.
But as with anything, it was worth mentioning.

cppwfs reviewed (approved):
LGTM 
Do you want me to merge and we start a new PR for the rest of the work, or continue with this PR?

artembilan committed:
Move JDBC components to new `inbound` or `outbound` packages, respectively

* Revise affected tests for the modern AssertJ API

artembilan committed:
Bring back old classes into root package as deprecated
and as extensions of moved classes to `inbound` & `outbound` packages

artembilan committed:
Move TCP & UDP adapters to respective new `tcp/inbound`,
`tcp/outbound`, `udp/inbound` or `udp/outbound` packages, respectively.
* Some code clean up for the affected classes and tests improvements
according to AssertJ API

artembilan committed:
Bring back moved TCP/IP classes to their old place
to preserve commit history.
And mark them as `@Deprecated`

artembilan committed:
Mention the package standardization in the `whats-new.adoc`

cppwfs commented on spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/package-info.java:
```diff
@@ -1,5 +0,0 @@
-/**
- * Base package for TCP Support.
- */
-@org.jspecify.annotations.NullMarked
```
Should this be readded for the deprecated classes?

cppwfs commented on src/reference/antora/modules/ROOT/pages/whats-new.adoc:
```diff
@@ -57,6 +57,13 @@ The `RetryStateGenerator` abstraction and its `SpelExpressionRetryStateGenerator
+All the modules now follow the standard package structure.
+The inbound and outbound components are declared in the `inbound` and `outbound` packages, respectively.
+The `MessageChannel` implementations in the `channel` package.
+Most of the modules have followed this rule from day first.
```
Update to say "from the first day"

cppwfs reviewed (changes requested):
Looks good.
Just a question and a quick doc fix and its ready to go.

artembilan committed:
* Fix sentence in `whats-new.adoc`
* Bring back `package-info.java` for the `org.springframework.integration.ip.tcp`
where we still have deprecated classes

cppwfs reviewed (approved):
LGTM.   

Want me to merge?
